woocommerce/woocommerce-ios
# Account Creation: Support login with security key

Part of: #10590 
Please also review https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/pull/798.


## What

In #11047, we switched to using custom UI for the login flow from account creation. This flow has yet to support login security key.

This PR adds support for security key through the custom UI following the [implementation](https://github.com/wordpress-mobile/WordPressAuthenticator-iOS/blob/trunk/WordPressAuthenticator/Unified%20Auth/View%20Related/2FA/TwoFAViewController.swift) in the WPAuthenticator library.

## Why
Security key login is required for A8C accounts, so this is needed to unblock A8C and any users who use security key for their WPCom authentication.

## How
- Added a new button on the 2FA login view to log in with a security key if needed.
- Updated the view model for the 2FA login view to follow the implementation in the authenticator library:
  - If nonce info is sent in the response of the password login step, follow a new method for logging in that includes nonce.
  - If the nonce in the nonce info is empty, display a generic error message. This signals that a security key is required.
  - To log in with a security key, the user needs to check their biometric with the device.

## Testing instructions

- Set up a security key for your A8C account following the guide in p91TBi-aIl-p2 to test with the iOS simulator if needed.
- Log out of the app if needed.
- On the prologue screen, select Create a new store.
- Enter the email address and password of your A8C account.
- Notice that the presented 2FA screen has a CTA for security login.
- Enter your 2FA code, notice that an error alert is displayed about the invalid code.
- Tap Log in with a security key to log in with your biometric data.
- Notice that after the login succeeds, the store creation flow is displayed.

Please feel free to also test with the login through the Jetpack setup flow again following the steps in https://github.com/woocommerce/woocommerce-ios/pull/9113 to make sure that everything still works properly.

## Screenshots


https://github.com/woocommerce/woocommerce-ios/assets/5533851/e0a4fa8c-063b-4ab8-a8bb-0908a128ef61



---
